enrichments/trace/config/config.go

```diff
@@ -0,0 +1,44 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package config
+
+// Config configures the enrichment attributes produced.
+type Config struct {
+	Transaction ElasticTransactionConfig `mapstructure:"elastic_transaction"`
+	Span        ElasticSpanConfig        `mapstructure:"elastic_span"`
+}
+
+// ElasticTransactionConfig configures the enrichment attributes for the
+// spans which are identified as elastic transaction.
+type ElasticTransactionConfig struct {
+	Type         AttributeConfig `mapstructure:"type"`
+	Result       AttributeConfig `mapstructure:"result"`
+	EventOutcome AttributeConfig `mapstructure:"event_outcome"`
+}
+
+// ElasticSpanConfig configures the enrichment attributes for the spans
+// which are NOT identified as elastic transaction.
+type ElasticSpanConfig struct {
+	EventOutcome  AttributeConfig `mapstructure:"event_outcome"`
+	ServiceTarget AttributeConfig `mapstructure:"service_target"`
+}
+
+// AttributeConfig is the configuration options for each attribute.
+type AttributeConfig struct {
+	Disabled bool `mapstructure:"disabled"`
```

Comment:
When it comes to the collector config, I think it may be a bit easier to understand if we use `enabled: true|false`, with the default being enabled for all attributes.

Reply:
I debated a bit on this (enabled vs disabled). If we choose enabled then we would have to provide something like `AllEnabledConfig` that can be used in the processor and be careful when we add new attributes. That is also okay. Do you think it makes sense to use `enabled` all the way (here in the lib as well as in the processor)?

Reply:
I have updated the code to use `enabled` as the config and also provided a method to get a configuration with all enrichments enabled.